### PR TITLE
Update Box2D to version 2.4.1

### PR DESCRIPTION
**Pull request by marmichalski**
~~This should be fine, but would be great if someone could run the premake build. I am creating a cmake build env and that's where I have tested these changes.~~ There is quite a few whitelines changes in touched files, I did not try to resolve them in any way, as unified line endings are my preference :P

~~Edit: or actually I probably forgot to update premake. I'll update in a few.~~

Done. It compiles & runs. ~~I have not updated the Visual Studio project as I have no way to test the changes - maybe just updating the files list should be enough :thinking:~~

image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -43,13 +43,12 @@ builddir:
...
-	tar xzf premake-5.0.0-alpha14-linux.tar.gz && \
-	rm premake-5.0.0-alpha14-linux.tar.gz)
-
-box2d: premake
-	.build/premake5 --file=third_party/Box2D/premake5.lua gmake
-	make -C third_party/Box2D/Build config=debug_x86_64 -j$(CPUS)
-	make -C third_party/Box2D/Build config=release_x86_64 -j$(CPUS)
-
-
+	wget https://github.com/premake/premake-core/releases/download/v5.0.0-beta1/premake-5.0.0-beta1-linux.tar.gz && \
```
Comment (marmichalski) [inline Makefile:46]: Updated premake also :shrug:

File: Makefile [hunk 2]
```diff
@@ -43,13 +43,12 @@ builddir:
 
 premake: builddir
 	test -e .build/premake5 || (cd .build && \
-	wget https://github.com/premake/premake-core/releases/download/v5.0.0-alpha14/premake-5.0.0-alpha14-linux.tar.gz && \
-	tar xzf premake-5.0.0-alpha14-linux.tar.gz && \
-	rm premake-5.0.0-alpha14-linux.tar.gz)
-
-box2d: premake
-	.build/premake5 --file=third_party/Box2D/premake5.lua gmake
```
Comment (marmichalski) [inline Makefile:51]: box has moved to cmake

File: src/PhysicsManager.cpp [hunk 3]
```diff
@@ -569,7 +569,7 @@ void PhysicsManager::HandleCollision_ObjectWithMap(b2Fixture* objectFixture, b2F
     float height = gGameMap.GetHeightAtPosition(gameObject->mPhysicsBody->GetPosition());
     int mapLayer = (int) (Convert::MetersToMapUnits(height) + 0.5f);
 
-    b2FixtureData_map fxdata = mapFixture->GetUserData();
+    b2FixtureData_map fxdata = (b2FixtureData_map*) mapFixture->GetUserData().pointer;
```
Comment (diath) [inline src/PhysicsManager.cpp:572]:
```suggestion
    b2FixtureData_map fxdata = static_cast<b2FixtureData_map*>(mapFixture->GetUserData().pointer);
```

File: src/PhysicsManager.cpp [hunk 4]
```diff
@@ -505,59 +505,59 @@ bool PhysicsManager::ShouldCollide_ObjectWithMap(b2Contact* contact, b2Fixture*
 
     // todo: this is temporary implementation
 
-    b2FixtureData_map fxdata = mapFixture->GetUserData();
+    b2FixtureData_map fxdata = (b2FixtureData_map*) mapFixture->GetUserData().pointer;
```
Comment (diath) [inline src/PhysicsManager.cpp:508]:
```suggestion
    b2FixtureData_map fxdata = static_cast<b2FixtureData_map*>(mapFixture->GetUserData().pointer);
```

**Review by diath — CHANGES_REQUESTED** [review]

**Comment by codenamecpp** [comment]
Oh, I didn't expect that. Nice work, thanks, @marmichalski !

As for 'static_cast' over 'c-style' cast, I haven't decided yet what I prefer more, usually I'm too lazy to write ***_cast<> etc but in general, I don't mind.

**Comment by marmichalski** [comment]
> Oh, I didn't expect that. Nice work, thanks, @marmichalski !
> 
> As for 'static_cast' over 'c-style' cast, I haven't decided yet what I prefer more, usually I'm too lazy to write ***_cast<> etc but in general, I don't mind.

Hey, no problem. I have updated the visual studio project for box2d, the `src` dir is needed in the additional include paths because there is, I believe, not correctly included file in one of the source files. I will open up a PR in Box2D and see what they think - if it will be accepted, the additional include path could be removed, if not, will have to stay with us :P

I am not updating the casts, because I see no reason to. I can do that, if that's what you prefer, @codenamecpp, but I see you don't mind too :)

Edit: Said box2d PR - https://github.com/erincatto/box2d/pull/702

**Pull request merged**